plugins/cost-insights/src/components/CostInsightsHeader/CostInsightsHeader.tsx

```diff
@@ -16,16 +16,15 @@
 
 import React from 'react';
 import { Typography } from '@material-ui/core';
+import useAsync from 'react-use/lib/useAsync';
 import { useCostInsightsStyles } from '../../utils/styles';
 import { Group } from '../../types';
-import {
-  identityApiRef,
-  ProfileInfo,
-  useApi,
-} from '@backstage/core-plugin-api';
+import { identityApiRef, useApi } from '@backstage/core-plugin-api';
 
-function name(profile: ProfileInfo | undefined): string {
-  return profile?.displayName || 'Mysterious Stranger';
+function useDisplayName(): string {
+  const identityApi = useApi(identityApiRef);
+  const state = useAsync(() => identityApi.getProfileInfo(), [identityApi]);
+  return state.value?.displayName || 'Mysterious Stranger';
```

Comment:
this will show `Mysterious Stranger` (for one tick, or longer depending on the identity impl) while loading, maybe better with the empty string at that point